src/Data/SongDifficulty.cpp

```diff
@@ -22,12 +23,21 @@ namespace SongDetailsCache {
     }
 
     bool SongDifficulty::ranked() const noexcept {
-        return stars > 0 && song().rankedStatus == RankedStatus::Ranked;
+        return this->rankedBL() || this->rankedSS();
     }
 
+    bool SongDifficulty::rankedBL() const noexcept {
+        return starsBL > 0 && hasFlags(song().rankedStates, RankedStates::BeatleaderRanked);
+    }
+
+    bool SongDifficulty::rankedSS() const noexcept {
+        return starsSS > 0 && hasFlags(song().rankedStates, RankedStates::ScoresaberRanked);
+    }
+
+    [[deprecated("This function is deprecated. It only works for scoresaber ranked difficulties and is not accurate.")]]
```

Comment:
is it really deprecated though? it's made to not be accurate, and pp is strictly scoresaber, not BL anyways